controller/api/public/prometheus.go

```diff
@@ -64,35 +64,53 @@ func (s *grpcServer) queryProm(ctx context.Context, query string) (model.Vector,
 }
 
 // add filtering by resource type
-// note that metricToKey assumes the label ordering (namespace, name)
 func promGroupByLabelNames(resource *pb.Resource) model.LabelNames {
 	names := model.LabelNames{namespaceLabel}
+	if resource.Type == k8s.Namespace {
+		return names
+	}
 
-	if resource.Type != k8s.Namespace {
-		names = append(names, promResourceType(resource))
+	if resource.Type == k8s.All {
+		for _, kind := range k8s.StatAllWorkloadKinds {
```

Comment:
This loop only iterates through k8s workload kinds like deployment, statefulset, job etc. Traffic split and authority kinds are handled outside of `k8s.All`. See changes below: https://github.com/linkerd/linkerd2/blob/2cf17da3d39befd5cc9531be708fdf73d3fff06a/controller/api/public/stat_summary.go#L107-L135